packages/types-cloudflare-worker/src/global.ts

```diff
@@ -150,38 +150,171 @@ export interface CloudflareWorkerGlobalKVPatch {
  */
 export interface CloudflareRequestAttributes extends CloudflareRequestFeatures {
   /**
-   * The TLS version used on the connection to Cloudflare.
+   * ASN of the incoming request. (e.g. 395747)
    */
-  readonly tlsVersion: string;
```

Comment:
With all of these fields now being optional, won't this require all fields to have their existence checked first before being used?

```typescript
if (event.request.cf.tlsVersion && event.request.cf.tlsVersion.includes('1.')) {
  ...
}
```

Reply:
Yes, that's my concern too. I've changed the test worker code to check for `null`, but wish I didn't have to.

Other options to reduce development hassles:
* Make a `makeRequestCf()` function accepting all optional values, but returning an object of required read-only values.
* Provide complete object which could be modified then cast to the `Request.cf` object. Sounds way too hacky.
* Force developers to copy/paste the entire `testIncomingRequest` object into their projects. Definitely worst option, but maybe still the best?

Reply:
Which of these four does anyone think is the best or even least worst?

Reply:
I feel pretty strongly that these fields should _not_ be optional, especially if they are all required/defined from CF's perspective. I think these should remain required, and we should add a helper function to help ease testing. (although i would personally suggest `makeCfRequest` over `makeRequestCf`)

Reply:
Two votes! Democracy works!

Reply:
Implemented 4 adding a `makeCloudflareWorkerRequest()` to the mocking package.